sky/shell/platform/ios/accessibility_bridge.h

```diff
@@ -0,0 +1,177 @@
+
+typedef scoped_refptr<Node> NodePtr;
+
+// Class that listens for updates to the semantic tree from Dart code and issues
+// corresponding updates to the FlutterView's accessibility elements.
+//
+// The bridge is owned by the FlutterView that created it. It maintains a raw
+// pointer back to the view to enable bidirectional communication with the view
+// without introducing a circular reference. Since the strong binding herein may
+// destroy the bridge, the FlutterView maintains its ownership via a weak
+// pointer reference.
+class AccessibilityBridge final : public semantics::SemanticsListener {
+ public:
+  AccessibilityBridge(FlutterView*, semantics::SemanticsServer*);
+  ~AccessibilityBridge() override;
+
+  void UpdateSemanticsTree(mojo::Array<semantics::SemanticsNodePtr>) override;
+
+  base::WeakPtr<AccessibilityBridge> AsWeakPtr();
+
+ private:
+  typedef std::map<long, NodePtr> NodeMap;
+  friend class Node;
+
+  NodePtr UpdateNode(const semantics::SemanticsNodePtr& node);
+  void RemoveNode(NodePtr node);
+
+  NSArray* CreateAccessibleElements() const NS_RETURNS_RETAINED;
+
+  // See class docs above about ownership relationship
+  FlutterView* view_;
+  NodeMap nodes_;
+
+  mojo::StrongBinding<semantics::SemanticsListener> binding_;
+
+  base::WeakPtrFactory<AccessibilityBridge> weak_factory_;
+
+  DISALLOW_COPY_AND_ASSIGN(AccessibilityBridge);
+};
+
+struct Flags {
+  Flags()
+      : can_be_tapped(false),
+        can_be_long_pressed(false),
+        can_be_scrolled_horizontally(false),
+        can_be_scrolled_vertically(false),
+        has_checked_state(false),
+        is_checked(false) {}
+
+  // Assigns the specified flags to this struct iff it is non-null.
+  // It is important not to update our flags when the semantic flags
+  // are null, because a null update from Dart means "nothing to update"
+  Flags& operator=(const semantics::SemanticFlagsPtr& other) {
+    if (!other.is_null()) {
+      can_be_tapped = other->canBeTapped;
+      can_be_long_pressed = other->canBeLongPressed;
+      can_be_scrolled_horizontally = other->canBeScrolledHorizontally;
+      can_be_scrolled_vertically = other->canBeScrolledVertically;
+      has_checked_state = other->hasCheckedState;
+      is_checked = other->isChecked;
+    }
+    return *this;
+  }
+
+  bool can_be_tapped : 1;
+  bool can_be_long_pressed : 1;
+  bool can_be_scrolled_horizontally : 1;
+  bool can_be_scrolled_vertically : 1;
+  bool has_checked_state : 1;
+  bool is_checked : 1;
+};
+
+struct Strings {
+  // Assigns the specified strings to this struct iff it is non-null
+  // It is important not to update our strings when the semantic strings
+  // are null, because a null update from Dart means "nothing to update"
+  Strings& operator=(const semantics::SemanticStringsPtr& other) {
+    if (!other.is_null()) {
+      if (!other->label.is_null()) {
+        label = other->label.get();
+      }
+    }
+    return *this;
+  }
+
+  std::string label = "";
```

Comment:
There's no need for this explicit initialization.  `std::string` defaults to the empty string.